src/App.tsx

```diff
@@ -8,55 +8,131 @@ import { PostsList } from './components/PostsList';
+          setPosts(formatPosts(res as PostResponse[]));
+        })
+        .catch(() => {
+          setIsPostsError(true);
+        })
+        .finally(() => {
+          setIsPostsLoading(() => false);
+        });
+    }
+  }, [user]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  user={user}
+                  setUser={setUser}
+                  users={users}
+                  setIsPostsLoading={setIsPostsLoading}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
-
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
-
-                <PostsList />
+                {isPostsLoading && <Loader />}
+                {!user && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
+                {!isPostsLoading && user && (
+                  <>
+                    {posts ? (
+                      <PostsList
+                        posts={posts}
+                        selectedPost={selectedPost}
+                        setSelectedPost={setSelectedPost}
+                        setIsNewCommentForm={setIsNewCommentForm}
+                        setIsCommentsError={setIsCommentsError}
+                      />
+                    ) : (
+                      <Notification
+                        title="No posts yet"
+                        dataCy="NoPostsYet"
+                        isImportant={false}
+                      />
+                    )}
+                  </>
+                )}
               </div>
             </div>
           </div>
-
-          <div
-            data-cy="Sidebar"
-            className={classNames(
-              'tile',
-              'is-parent',
-              'is-8-desktop',
-              'Sidebar',
-              'Sidebar--open',
-            )}
-          >
-            <div className="tile is-child box is-success ">
-              <PostDetails />
+          {selectedPost && (
+            <div
+              data-cy="Sidebar"
+              className={classNames(
+                'tile',
+                'is-parent',
+                'is-8-desktop',
+                'Sidebar',
+                { 'Sidebar--open': selectedPost },
```

Comment:
Remove the unnecessary 'selectedPost' check within the className object, as it is already checked at line 130.